src/pow.cpp

```diff
@@ -92,6 +92,8 @@ std::string Pow::__str__() const
     std::ostringstream o;
     if (is_a<Add>(*base_)) {
         o << "(" << *base_ << ")";
+    } else if (is_a<Mul>(*base_)) {
+        o << "(" << *base_ << ")";
```

Comment:
This may be unrelated to the PR, but we should also check for `Pow`, since `(x**y)**z` prints as `x^y^z`